README.md

```diff
@@ -1,2 +1,215 @@
+  console.log('done dispatching sequence!')
+});
+```
+
+## Installation
+```bash
+npm install --save redux-phunk
+```
+Then use [applyMiddleware](http://redux.js.org/docs/api/applyMiddleware.html) to use redux-phunk.
+```js
+import reducer from './reducer.js';
+import { createStore, applyMiddleware } from 'redux';
+import phunk from 'redux-phunk';
+
+const store = createStore(reducer, applyMiddleware(phunk));
+
+store.dispatch(myAction());
+```
+
+## Phunks?? That's not even a word!
+A "phunk" is just an object with `name` and `phunk` properties. The `phunk` property is a function that returns an asynchronous value, such as a Promise or a (single-valued) Observable. Examples:
+
+```js
+// es2016
+{
+  name: "getUser",
+  async phunk() {
+    return await fetch('/api/users/1').then(response => response.json());
+  },
+}
+```
+```js
+// es6 + Observables
+{
+  name: "getTypeaheadOptions",
+  phunk() {
+    return Observable.just([1, 2, 3]);
+  }
+}
+```
+```js
+// es5
+{
+  name: "saveComment",
+  phunk: function() {
+    return new Promise(function(resolve, reject) {
+      doAjax(function(err) {
+        if (err) {
+          reject(err);
+        } else {
+          resolve()
+        }
+      })
+    }
+  }
+}
+```
+
+## Putting async values into vanilla redux actions
+Running asynchronous code as a "phunk" is great, but in order to actually use the asynchronous value, redux-phunk provides the `asyncValue` function. This is used inside of vanilla redux actions so that the asynchronous value is passed to the reducer, where it will be transformed and handled properly. In the following example, the /api/books/:id api will first be called. Once the ajax request is completed, GOT_BOOK will be dispatched with `newBook` set to whatever the api returned for the book:
+
+```js
+import { asyncValue } from 'redux-phunk';
+
+function getBook(id) {
+  return [
+    {
+      name: 'book',
+      async phunk() {
+        return await fetch(`/api/books/${id}`)
+          .then(response => response.json());
+      }
+    },
+    {
+      type: 'GOT_BOOK',
+      newBook: asyncValue('book'),
+    },
+  ]
+}
+
+store.dispatch(getBook());
+```
+
+## Using async values inside of another phunk
+If you need to reuse asynchronous values across multiple phunks, you can call the `valueOf` function provided as the first argument to each phunk.
+```js
+import { asyncValue } from 'redux-phunk';
+
+function getUserPreferences(userId) {
+  return [
+    {
+      name: 'user',
+      async phunk() {
+        return await fetch(`/api/users/${userId}`)
+          .then(response => response.json())
+      },
+    },
+    {
+      name: 'userPreferences',
+      async phunk(valueOf) {
+        const userPreferencesId = valueOf('user').preferencesId;
+        return await fetch(`/api/user-preferences/${userPreferencesId}`)
+          .then(response => response.json())
+      }
+    },
+    {
+      type: 'NEW_USER_PREFERENCES',
+      preferences: asyncValue('userPreferences'),
```

Comment:
If two separate phunk sequences are fired at the same time, `asyncValue` does *not* allow you to pull values from a different sequence.

See [what async value actually does](https://github.com/CanopyTax/redux-phunk/blob/start/src/redux-phunk.js#L21), where async values are given a [concrete value](https://github.com/CanopyTax/redux-phunk/blob/start/src/redux-phunk.js#L88), and where the concrete value comes from ([see](https://github.com/CanopyTax/redux-phunk/blob/start/src/redux-phunk.js#L93) and [here](https://github.com/CanopyTax/redux-phunk/blob/start/src/redux-phunk.js#L6)).